src/Libraries/DSCPython/DSCPythonCodeCompletionProviderCore.cs

```diff
@@ -0,0 +1,1270 @@
+        private string GetLastNameSpace(string text)
+        {
+            return MATCH_LAST_NAMESPACE.Match(text.Trim('.').Trim()).Value;
+        }
+
+        /// <summary>
+        /// Returns the first possible type name from the type's declaration line.
+        /// </summary>
+        /// <param name="line"></param>
+        /// <returns></returns>
+        private static string GetFirstPossibleTypeName(string line)
+        {
+            var match = MATCH_VALID_TYPE_NAME_CHARACTERS_ONLY.Match(line);
+            string possibleTypeName = match.Success ? match.Value : "";
+            return possibleTypeName;
+        }
+
+        /// <summary>
+        /// Removes any docstring characters from the source code
+        /// </summary>
+        /// <param name="code"></param>
+        /// <returns></returns>
+        private string StripDocStrings(string code)
+        {
+            var matches = TRIPLE_QUOTE_STRINGS.Split(code);
+            return String.Join("", matches);
+        }
+
+        /// <summary>
+        /// Detect all library references given the provided code
+        /// </summary>
+        /// <param name="code">Script code to search for CLR references</param>
+        /// <returns></returns>
+        private List<string> FindClrReferences(string code)
+        {
+            var statements = new List<string>();
+            foreach (var line in code.Split(new[] { '\n', ';' }))
+            {
+                if (line.Contains("clr.AddReference"))
+                {
+                    statements.Add(line.Trim());
+                }
+            }
+
+            return statements;
+        }
+
+        /// <summary>
+        /// Check if a full type name is found in one of the known pre-loaded assemblies and return the type
+        /// </summary>
+        /// <param name="name">a full type name</param>
+        /// <returns></returns>
+        private Type TryGetTypeFromFullName(string name)
+        {
+            foreach (var asName in knownAssemblies)
+            {
+                Type foundType = Type.GetType(String.Format("{0},{1}", name, asName));
+                if (foundType != null)
+                {
+                    return foundType;
+                }
+            }
+
+            return null;
+        }
+
+        //Even though the following region is full of private implementation details, these
+        // back the old IronPythonCompletionProvider class
+        //!!!- do not modify these signatures until that class is removed.
+        //We do not know if anyone was using that class, but we needed to remove the compile
+        //time references between PythonNodeModels and DSIronPython to support dynamically loading
+        //python versions.
+
+        //note that the public members below are not really public (this is an internal class)
+        //but must be marked that way to satisfy the legacy interface
+        #region BACKING LEGACY CLASS DO NOT MODIFY UNTIL 3
```

Comment:
I think I wrote this comment, so it seems this was copy pasted from the other file... yeah we should be able to get rid of all of this unless it's used here?

Reply:
They are being used for now because we do not have a common class for both of the CompletionProviderCore classes. My thinking was if only one of the IronPython/CPython is present, we would still need these helper functions.